common.tf

```diff
@@ -240,6 +242,19 @@ module "dr2_ingest_step_function" {
   }
 }
 
+module "ingest_state_bucket" {
+  source      = "git::https://github.com/nationalarchives/da-terraform-modules//s3"
+  bucket_name = local.ingest_state_bucket_name
+  bucket_policy = templatefile("./templates/s3/lambda_access_bucket_policy.json.tpl", {
+    lambda_role_arns = jsonencode([
+      module.dr2_ingest_asset_opex_creator_lambda.lambda_role_arn,
+      module.dr2_ingest_folder_opex_creator_lambda.lambda_role_arn,
+      module.dr2_ingest_asset_reconciler_lambda.lambda_role_arn
+    ]),
+    bucket_name = local.ingest_state_bucket_name
+  })
```

Comment:
Only the Mapper Lambda and Step Function itself will need access to the bucket (read/write and read respectively).

Whilst not required as the bucket is in the same account as the resources, the bucket should also be added to the permission policy of the roles used by the Lambda and Step Function. Normally, resource policies are reserved for managing service access to a resource or cross account.